src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageService.cs

```diff
@@ -94,29 +93,98 @@ private TemporaryStorageService(IWorkspaceThreadingService? workspaceThreadingSe
         _textFactory = textFactory;
     }
 
-    public ITemporaryTextStorageInternal CreateTemporaryTextStorage()
-        => new TemporaryTextStorage(this);
+    ITemporaryStorageTextHandle ITemporaryStorageServiceInternal.WriteToTemporaryStorage(SourceText text, CancellationToken cancellationToken)
+        => WriteToTemporaryStorage(text, cancellationToken);
 
-    public TemporaryTextStorage AttachTemporaryTextStorage(
-        string storageName, long offset, long size, SourceHashAlgorithm checksumAlgorithm, Encoding? encoding, ImmutableArray<byte> contentHash)
-        => new(this, storageName, offset, size, checksumAlgorithm, encoding, contentHash);
+    async Task<ITemporaryStorageTextHandle> ITemporaryStorageServiceInternal.WriteToTemporaryStorageAsync(SourceText text, CancellationToken cancellationToken)
+        => await WriteToTemporaryStorageAsync(text, cancellationToken).ConfigureAwait(false);
 
-    ITemporaryStorageHandle ITemporaryStorageServiceInternal.WriteToTemporaryStorage(Stream stream, CancellationToken cancellationToken)
+    public TemporaryStorageTextHandle WriteToTemporaryStorage(SourceText text, CancellationToken cancellationToken)
+    {
+        var memoryMappedInfo = WriteToMemoryMappedFile();
+        var identifier = new TemporaryStorageIdentifier(memoryMappedInfo.Name, memoryMappedInfo.Offset, memoryMappedInfo.Size);
+        return new(this, memoryMappedInfo.MemoryMappedFile, identifier, text.ChecksumAlgorithm, text.Encoding, text.GetContentHash());
+
+        MemoryMappedInfo WriteToMemoryMappedFile()
```

Comment:
this code was inlined from the old TemporaryTextStorage type (which was deleted).